src/extractcode/extract.py

```diff
@@ -97,7 +97,21 @@
 ExtractEvent = namedtuple('ExtractEvent', 'source target done warnings errors')
 
 
-def extract(location, kinds=extractcode.default_kinds, recurse=False):
+def extract(location, kinds=extractcode.default_kinds, recurse=False, replace_originals=False):
+    events = list(extract_files(location, kinds, recurse))
+    if replace_originals:
+        for xevent in reversed(events):
+            if xevent.done:
+                source = xevent.source
+                target = xevent.target
+                if TRACE:
+                    logger.debug('extract:replace_originals: replace %(source)r by %(target)r' % locals())
+                fileutils.delete(source)
+                fileutils.copytree(target, source)
+                fileutils.delete(target)
```

Comment:
Just FYI and nitpicking: You could have been using the standard library `os.rename/os.renames` instead rather than the `fileutils` copytree/delete. https://docs.python.org/3/library/os.html?highlight=os%20rename#os.rename 

On linux at least `os.rename` is using the Linux kernel `rename` "atomic" syscall behind the scenes which is best as it will pass or fail at once IMHO

If you want to go that route that would become:
```
                fileutils.delete(source)
                os.rename(target, source)
 ``` 

You would need to also add an `import os`  at the top https://github.com/nexB/scancode-toolkit/pull/1893/files#diff-16d915471be787f821dcd1d38891c2bdR32 

Again no change required, this is more of a FYI

Reply:
I also had that code, but I thought it looks more consistent if I use your utils

Reply:
Both work. The fileutils ways have been heavily tested on all OSes too. So Let's keep it.